network/vif-route-qubes

```diff
@@ -118,32 +118,51 @@ domid=${domid/.*/}
 #  32752 is max XID aka domid
 metric=$(( 32752 - domid ))
 
-
+# add anti-spoofing rules before enabling the interface
 if [ "${ip}" ]; then
     # If we've been given a list of IP addresses, then add routes from dom0 to
     # the guest using those addresses.
     for addr in ${ip};
     do
-        if [[ "$addr" = *:* ]] && [[ "$ipv6_disabled" = 1 ]]; then
-            log error "Cannot set IPv6 route to ${addr}, IPv6 disabled in the kernel"
-            continue
-        fi
-        ${cmdprefix} ip route "${ipcmd}" "${addr}" dev "${vif}" metric "$metric"
         if [[ "$addr" = *:* ]]; then
             ipt=ip6tables-restore
         else
             ipt=iptables-restore
         fi
-        echo -e "*raw\\n$iptables_cmd -i ${vif} ! -s ${addr} -j DROP\\nCOMMIT" | \
+        printf '%s\n' "*raw" \
+            "$iptables_cmd -i ${vif} ! -s ${addr} -j DROP" \
+            "$iptables_cmd ! -i vif+ -s ${addr} -j DROP" \
+            "COMMIT" | \
             ${cmdprefix} $ipt --noflush $ipt_arg
-
-        network_hooks "${command}" "${vif}" "${addr}"
+        if [[ "$command" = 'online' ]]; then
+            ip neighbour "${ipcmd}" to "${addr}" \
+                dev "${vif}" lladdr 00:16:3e:5e:6c:6c:00 nud permanent
```

Comment:
`6c` is duplicated. But also, this will break with custom MAC address (`mac` VM's property is writable). I do see `mac` property in xenstore for an interface. I'm not sure if it's already retrieved into some variable, but you can read it from xenstore (seems like vif-nat script does that, for example).

Reply:
Fixed